src/connector/src/source/datagen/mod.rs

```diff
@@ -29,7 +30,7 @@ use crate::source::SourceProperties;
 pub const DATAGEN_CONNECTOR: &str = "datagen";
 
 #[serde_as]
-#[derive(Clone, Debug, Deserialize)]
+#[derive(Clone, Debug, Deserialize, MarkRedaction)]
```

Comment:
just for confirmation, the trait happens when calling Display and Debug? Will it influent the contents in etcd?

Reply:
MarkRedaction::marks need be called explicitly, currently only in `try_redact_statement` which redacts a `Statement`. We don't redact the `*Properties` structs.

> All those *Properties that implement MarkRedaction can tell that info via MarkRedaction::marks, i.e. properties names that need redaction.

https://github.com/risingwavelabs/risingwave/pull/14193#issuecomment-1869840023